gobdd.go

```diff
@@ -21,9 +21,10 @@ import (
 
 // Holds all the information about the suite (options, steps to execute etc)
 type Suite struct {
-	t       *testing.T
-	steps   []stepDef
-	options SuiteOptions
+	t           tester
+	steps       []stepDef
+	options     SuiteOptions
+	stepsErrors []error
```

Comment:
Steps could be validated in `Run` as well. What's the reason for caching errors?

Reply:
to make the `Run` function as small as possible. There's no another way of adding steps than using `AddStep*`.

The second reason why I did it is that (maybe in the future) we'll want to add more checks before calling the `Run`. Then, the only thing we'll need to do is to rename it.

Reply:
Hm...I still think it might better be to run checks in the beginning of `Run`. Eg. `validateSteps()`. It would keep both `Run` and `AddStep*` clean.

Reply:
I think it's not so simple. When working with regex's, we have to compile them. If we do it in `AddStep` then we have to or use MustCompile (which throws a panic) or regular Compile but we have to store the error somewhere. It means, there have to be some `if err != nil` in the function. 

Or... store strings (not compiled) and try to compile them in the `validateSteps()` function. But IMO it would complicate things even more.

By doing it this way, we're at least consistent with the behaviour.

Reply:
Ah, okay, I see. Well, it's an internal detail, so it should be fine.